src/oneD/StFlow.cpp

```diff
@@ -347,10 +382,25 @@ void StFlow::evalResidual(double* x, double* rsd, int* diag,
             // Continuity. This propagates information right-to-left, since
             // rho_u at point 0 is dependent on rho_u at point 1, but not on
             // mdot from the inlet.
+            if(m_strainRateEq == false)
             rsd[index(c_offset_U,0)] =
                 -(rho_u(x,1) - rho_u(x,0))/m_dz[0]
                 -(density(1)*V(x,1) + density(0)*V(x,0));
-
+			else
+			rsd[index(c_offset_U,0)] = -(rho_u(x,1) - rho_u(x,0))/m_dz[0]
+				-0.5*(lambda(x,1)*density(1)*V(x,1) + lambda(x,0)*density(0)*V(x,0));
+
+            // Control to switch between regular Cantera, one-point or two-point control methods
+            if (m_twoPointControl == false && m_onePointControl == false)  
```

Comment:
There's no need to compare booleans to the `true` and `false` literals -- this is more simply written as
```c++
if (!m_twoPointControl && !m_onePointControl)
```